src/data/sym/basic.lean

```diff
@@ -150,4 +159,111 @@ instance inhabited_sym' [inhabited α] (n : ℕ) : inhabited (sym' α n) :=
 
 end inhabited
 
+instance has_zero {α : Type*} : has_zero (sym α 0) := ⟨⟨0, rfl⟩⟩
+instance has_emptyc {α : Type*} : has_emptyc (sym α 0) := ⟨0⟩
+
+instance subsingleton {α : Type*} {n : ℕ} [g : subsingleton α] : subsingleton (sym α n) :=
+⟨begin
+  unfreezingI { cases g },
+  intros,
+  rcases a with ⟨c, d⟩,
+  rcases b with ⟨a, b⟩,
+  simp only [subtype.mk.inj_eq],
+  induction a using multiset.case_strong_induction_on with k hk wa generalizing n c,
+  { rw [b.symm, multiset.card_zero] at d, exact multiset.card_eq_zero.mp d },
+  { cases n,
+    { exact false.elim (multiset.cons_ne_zero (multiset.card_eq_zero.mp b)) },
+    { classical,
+      exact if s : c = 0 then begin
```

Comment:
I think it's more idiomatic to write `by_cases s : c = 0` instead of `exact if s : c = 0 then .. else ..`.